compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -1529,11 +1529,18 @@ public BLangNode transform(NamedWorkerDeclarationNode namedWorkerDeclNode) {
         }
 
         // change default worker name
-        String workerName = namedWorkerDeclNode.workerName().text();
+        String workerName;
+        if (namedWorkerDeclNode.workerName().isMissing()) {
+            workerName = missingNodesHelper.getNextMissingNodeName(packageID);
+        } else {
+            workerName = namedWorkerDeclNode.workerName().text();
+        }
+
```

Comment:
Doing a fix at line no 1620 is another quick solution, but the above is a good fix to provide a better clarify for the flow.